tests/unit/test_document_revision.py

```diff
@@ -0,0 +1,523 @@
+        child,
+        sub_child,
+        second_parent,
+        second_child,
+    ]
+    second_child.root_revision_nodes = [
+        parent,
+        child,
+        sub_child,
+        second_parent,
+        second_child,
+    ]
+
+    parent.all_node_ids = [1, 2, 3, 4, 5]
+    child.all_node_ids = [1, 2, 3, 4, 5]
+    sub_child.all_node_ids = [1, 2, 3, 4, 5]
+    second_parent.all_node_ids = [1, 2, 3, 4, 5]
+    second_child.all_node_ids = [1, 2, 3, 4, 5]
+
+    parent.children_ids = [2, 5]
+    child.children_ids = [3]
+
+    parent.parent_ids = [1, 4]
+    second_parent.parent_ids = [1, 4]
+
+    child.children.append(sub_child)
+    parent.children.append(child)
+    parent.children.append(second_child)
+    second_child.children = []
+    second_parent.children = []
+
+    yield [parent, child, sub_child, second_parent, second_child]
+
+
+@mock.patch("google.cloud.documentai_toolbox.wrappers.document_revision.storage")
+def test_get_base_and_revision_bytes(mock_storage):
+    client = mock_storage.Client.return_value
+    mock_bucket = mock.Mock()
+    client.Bucket.return_value = mock_bucket
+
+    with open("tests/unit/resources/revisions/doc.dp.bp", "rb") as f:
+        doc_bytes = f.read()
+
+    with open("tests/unit/resources/revisions/pages_00001.dp.bp", "rb") as f:
+        pages_bytes = f.read()
+
+    with open("tests/unit/resources/revisions/rev_00000.dp.bp", "rb") as f:
+        rev0_bytes = f.read()
+
+    with open("tests/unit/resources/revisions/rev_00001.dp.bp", "rb") as f:
+        rev1_bytes = f.read()
+
+    with open("tests/unit/resources/revisions/rev_00002.dp.bp", "rb") as f:
+        rev2_bytes = f.read()
+
+    with open("tests/unit/resources/revisions/rev_00003.dp.bp", "rb") as f:
+        rev3_bytes = f.read()
+
+    with open("tests/unit/resources/revisions/rev_00004.dp.bp", "rb") as f:
+        rev4_bytes = f.read()
+
+    doc = mock.Mock(name=[])
+    doc.name = "gs://test-directory/1/doc.dp.bp"
+    doc.download_as_bytes.return_value = doc_bytes
+
+    pages_00001 = mock.Mock(name=[])
+    pages_00001.name = "gs://test-directory/1/pages_00001.dp.bp"
+    pages_00001.download_as_bytes.return_value = pages_bytes
+
+    rev_00000 = mock.Mock(name=[])
+    rev_00000.name = "gs://test-directory/1/rev_00000.dp.bp"
+    rev_00000.download_as_bytes.return_value = rev0_bytes
+
+    rev_00001 = mock.Mock(name=[])
+    rev_00001.name = "gs://test-directory/1/rev_00001.dp.bp"
+    rev_00001.download_as_bytes.return_value = rev1_bytes
+
+    rev_00002 = mock.Mock(name=[])
+    rev_00002.name = "gs://test-directory/1/rev_00002.dp.bp"
+    rev_00002.download_as_bytes.return_value = rev2_bytes
+
+    rev_00003 = mock.Mock(name=[])
+    rev_00003.name = "gs://test-directory/1/rev_00003.dp.bp"
+    rev_00003.download_as_bytes.return_value = rev3_bytes
+
+    rev_00004 = mock.Mock(name=[])
+    rev_00004.name = "gs://test-directory/1/rev_00004.dp.bp"
+    rev_00004.download_as_bytes.return_value = rev4_bytes
+
```

Comment:
This one is readable as is, but it's usually preferred to provide all initial values when creating an object.

Reply:
for some reason putting the path in `.name` in initialization causes an issue with `os.path.basename`